examples/rpc-custom-api/src/example.ts

```diff
@@ -12,13 +12,28 @@ import {
     createRpc,
     createSolanaRpcApi,
     DEFAULT_RPC_CONFIG,
+    isJsonRpcPayload,
     mainnet,
     RpcApi,
+    RpcTransport,
+    RpcTransportMainnet,
     SolanaRpcApiMainnet,
 } from '@solana/web3.js';
 
 const log = createLogger('Custom JSON RPC API');
 
+// This is a workaround that avoids the need to send RPC calls to mainnet during CI.
+function mockRpcTransportForCI(): RpcTransportMainnet {
+    const transport = createDefaultRpcTransport({ url: 'http://127.0.0.1:8899' });
+    return (async <TResponse>({ payload, signal }: Parameters<RpcTransport>[0]) => {
+        if (isJsonRpcPayload(payload) && payload.method === 'getAsset') {
+            const metadata = { name: 'USD Coin', symbol: 'USDC' };
+            return { result: { content: { metadata } } } as TResponse;
+        }
```

Comment:
I think it'd be worth verifying `params` here too, just to make sure we don't break anything in how they're passed through

I'd also throw an error or at least log the payload in the else case to help with debugging